packages/xstate-fsm/src/types.ts

```diff
@@ -86,10 +86,14 @@ export namespace StateMachine {
     assignment: Assigner<TContext, TEvent> | PropertyAssigner<TContext, TEvent>;
   }
 
-  export type Transition<TContext extends object, TEvent extends EventObject> =
-    | string
+  export type Transition<
+    TContext extends object,
+    TEvent extends EventObject,
+    TStateValue extends string
+  > =
+    | TStateValue
     | {
-        target?: string;
+        target?: TStateValue;
```

Comment:
What exactly does this solve for you? Could you share some code that is affected by this?

Reply:
@Andarist 
I decided that it would be better to show in dynamics. This file describes the state machine. The creation takes place elsewhere, here I wanted to show the development process. Please ask more questions if you need to explain in more detail what is happening in the video.

https://github.com/statelyai/xstate/assets/68888927/5c216593-56d0-43e8-9bf0-cd862b4f4039

```
import { StateMachine } from '@xstate/fsm';

export type TestContext = {
  readonly value: number;
};

export type TestStateValue = 'init' | 'loading' | 'idle' | 'process';

export type TestStateMachine = StateMachine.Config<
  TestContext,
  { type: 'NEXT' | 'STOP' | 'COMPLETE' },
  { value: TestStateValue; context: TestContext }
>;

export const createStateMachine = (): TestStateMachine => ({
  context: { value: 42 },
  initial: 'init',
  states: {
    init: {
      on: {
        COMPLETE: 'loading',
      },
    },
    loading: {
      on: {
        COMPLETE: { target: 'idle' },
      },
    },
    idle: {
      on: {
        NEXT: 'process',
      },
    },
    process: {
      on: {
        STOP: { target: 'idle' },
      },
    },
  },
});

```